controllers/users.js

```diff
@@ -1,11 +1,31 @@
 const User = require('../models/user');
+const UserInformation = require('../models/userinformation');
+const Joi = require("joi");
 
 const mapboxGeocoding = require('@mapbox/mapbox-sdk/services/geocoding');
 const mapboxToken = process.env.MAPBOX_TOKEN;
 const geocoder = mapboxGeocoding({ accessToken: mapboxToken });
 const axios = require('axios');
 const NE_BASE_URL = process.env.NEGOTIATION_ENGINE_BASE_URL || "http://localhost:5000";
 
+// Schema to validate inputs for `creating profile` - NOT USING THIS FOR NOW.
+const createUserSchema = Joi.object({
+	username: Joi.string().allow(null, ''),
+	email: Joi.string().allow(null, ''),
+	firstname: Joi.string().allow(null, ''),
+	surname: Joi.string().allow(null, ''),
+	phone: Joi.string().allow(null, ''),
+	address1: Joi.string().allow(null, ''),
+	address2: Joi.string().allow(null, ''),
+	postcode: Joi.string().allow(null, ''),
+	area: Joi.string().allow(null, ''),
+	country: Joi.string().allow(null, ''),
+	state: Joi.string().allow(null, ''),
+	description: Joi.string().allow(null, ''),
+	details: Joi.string().allow(null, '')
+  });

```

Comment:
Tror inte du behöver tillåta `null`, enligt [docs](https://joi.dev/api/?v=17.6.0#string) är "All keys are optional by default.", dvs antar den tillåter `undefined`. Men inte kört nå mycket på icke-required.

Reply:
Aha okej, men har ändrat så att den används nu iaf, kör med exakt samma som jag har där. Har också ändrat i funktionen "createEditPage" i controllern för users så att Nu om man går in och uppdaterar någon information om sig själv så är det endast det du har ändrat som uppdateras och alla andra fält behålls.